cardano-cli/src/Cardano/CLI/Run/Friendly.hs

```diff
@@ -131,21 +137,20 @@ friendlyTxOut (TxOut addr amount mdatum) =
           datum ShelleyBasedEraAllegra = []
           datum ShelleyBasedEraMary = []
           datum ShelleyBasedEraAlonzo = ["datum" .= renderDatum mdatum]
-      in object $ preAlonzo ++ datum sbe
+      in preAlonzo ++ datum sbe
   where
-   renderDatum :: TxOutDatum CtxTx era -> Aeson.Value
-   renderDatum TxOutDatumNone = Aeson.Null
-   renderDatum (TxOutDatumHash _ h) =
-     Aeson.String $ serialiseToRawBytesHexText h
-   renderDatum (TxOutDatum _ sData) =
-     scriptDataToJson ScriptDataJsonDetailedSchema sData
+    renderDatum :: TxOutDatum CtxTx era -> Aeson.Value
+    renderDatum TxOutDatumNone = Aeson.Null
+    renderDatum (TxOutDatumHash _ h) =
+      Aeson.String $ serialiseToRawBytesHexText h
+    renderDatum (TxOutDatum _ sData) =
+      scriptDataToJson ScriptDataJsonDetailedSchema sData
 
-
-friendlyStakeReference :: Crypto crypto => Shelley.StakeReference crypto -> Aeson.Value
+friendlyStakeReference :: StakeAddressReference -> (Text, Aeson.Value)
 friendlyStakeReference = \case
-  Shelley.StakeRefBase cred -> toJSON cred
-  Shelley.StakeRefNull -> Null
-  Shelley.StakeRefPtr ptr -> toJSON ptr
+  NoStakeAddress -> "stake reference" .= Null
+  StakeAddressByPointer ptr -> "stake reference" .= String (show ptr)
+  StakeAddressByValue cred -> friendlyStakeCredential "reference" cred
```

Comment:
What does the `"reference"` mean? All of the keys here should be the same i.e "stake address".

Reply:
To designate this stake address relation to the place where it is used without extra 1-key object.

Reply:
rewrote